orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -75,6 +76,9 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(
       rule_transformer->Register(make_unique<NonZeroShapeSetter>());
       rule_transformer->Register(make_unique<InsertSoftmaxCrossEntropyLossOutput>());
 
+      // Remove duplicate nodes. Must be applied before any recompute transformations.
+      transformers.emplace_back(onnxruntime::make_unique<CommonSubexpressionElimination>(compatible_eps));
```

Comment:
I think this still will be in conflict with when we enable recompute. The graph transformer manager will apply the transformations multiple times. 

So CommonSubexpressionElimination will eliminate the recompute nodes that was inserted in the last iteration.